include/zephyr/zephyr.h

```diff
@@ -7,14 +7,6 @@
 #ifndef ZEPHYR_INCLUDE_ZEPHYR_H_
 #define ZEPHYR_INCLUDE_ZEPHYR_H_
 
-/*
- * Applications can identify whether they are built for Zephyr by
- * macro below. (It may be already defined by a makefile or toolchain.)
- */
-#ifndef __ZEPHYR__
-#define __ZEPHYR__
-#endif
```

Comment:
The rationale for this is given in the commit https://github.com/zephyrproject-rtos/zephyr/commit/0c6df2d55cd934c20f9add4f3fda7d59a1fc4051 from almost 5 years ago:

> Currently it's defined by Makefile, so if an app uses it's own build system, it won't be
available.

This is no longer valid (an app must use the Zephyr's build system to maintain everyone's sanity), and `zephyr.h` is not a "must-include" header either, so it only makes sense to remove this now.